source/error.c

```diff
@@ -29,21 +29,15 @@ static void *s_global_error_context = NULL;
 static AWS_THREAD_LOCAL aws_error_handler_fn *tl_thread_handler = NULL;
 AWS_THREAD_LOCAL void *tl_thread_handler_context = NULL;
 
-#ifndef AWS_MAX_ERROR_SLOTS
-#    define AWS_MAX_ERROR_SLOTS 16
-#endif
-
 /* Since slot size is 00000100 00000000, to divide, we need to shift right by 10
  * bits to find the slot, and to find the modulus, we use a binary and with
- * 00000011 11111111 to find the index in that slot. The next three values
- * define those constants */
-#define AWS_ERROR_SLOT_SIZE 0x0400
-#define SLOT_DIV_SHIFT 0x0A
-#define SLOT_MASK 0x03FF
+ * 00000011 11111111 to find the index in that slot.
+ */
+#define SLOT_MASK (AWS_ERROR_ENUM_STRIDE - 1)
 
-static const int MAX_ERROR_CODE = AWS_ERROR_SLOT_SIZE * AWS_MAX_ERROR_SLOTS;
+static const int MAX_ERROR_CODE = AWS_ERROR_ENUM_STRIDE * AWS_PACKAGE_SLOTS;
```

Comment:
why unsigned below but signed here? I don't care either way, I'd just push for consistency.

Reply:
We're using signed integers for errors and unsigned integers for the other enums (log subject, statistics category).  I don't think it's necessarily inconsistent to have the bounds constants match the associated integer type rather than be the same across all.

Reply:
Fair enough. Ship it.